game-core/src/main/java/games/strategy/engine/chat/ChatMessageListener.java

```diff
@@ -6,7 +6,7 @@
 public interface ChatMessageListener {
   void addMessage(String message, PlayerName from);
 
-  void addMessageWithSound(String message, PlayerName from, String sound);
+  void addSlapMessage(String message, PlayerName from);
```

Comment:
Any thought to rename this to `addSlapEvent`? 
There is also a false consistency of the "add" prefix, `addStatusMessage` is badly named, since it's setting a status message. But for slap event, 'add' is not the worst verb, but I'm struck as it's not really a message but more a chat event.

Reply:
I wanted to stay consistent with `addMessage` and `addStatusMessage`, because in the end all of those "events" somehow add a message on a display or in a log or something.
I'd appreciate if you would open a PR with name proposals for all 3 methods that suit better in your opinion, but I think this discussion is out-of-scope for this PR.

Reply:
Sure, it's a false consistency though, it's being consistent with something that is incorrect. It would lead to something like  "addPlayerJoinedMessage", which is not a message. It's also a question what exactly are we adding to? It's an implementation detail that there is a chat history, and that could potentially be removed.

So, `addSlapEvent`, are you opposed to that? IMO it's probably better like "playerWasSlapped", or "slapEventOccurred", or "handleSlap", but fundamentally I'd prefer a distinction between messages and chat events (messages could be considered a subset, but not everything is a message, like a player joining, which should arguably be part of this interface)

Reply:
I got a bit carried away, "addStatusMessage" is incorrect, it's not adding, it's setting, and a status is not at all a message.

Reply:
Sorry, I was incorrect, 'addStatusMessage' is just confusing, as we have an important concept of a player status, and this 'addStatusMessage' has nothing to do with that. The 'addStatusMessage' is really just a generic chat event. Sigh..